drivers/overlay/encryption.go

```diff
@@ -475,6 +475,11 @@ func (d *driver) updateKeys(newKey, primary, pruneKey *key) error {
 		}
 	}
 
+	d.secMapWalk(func(rIPs string, spis []*spi) ([]*spi, bool) {
+		rIP := net.ParseIP(rIPs)
+		return updateNodeKey(lIP, aIP, rIP, spis, d.keys, newIdx, priIdx, delIdx), false
+	})
+
 	if (newKey != nil && newIdx == -1) ||
 		(primary != nil && priIdx == -1) ||
```

Comment:
If my understanding is correct, there is mismatch between driver.key and driver.secMap.node[].spi when an error happens here. The driver.key is already appended in line 466, and that is only change to the driver state.

I wonder if it is better to revert driver.key state when there is error here? That way driver.key and driver.secMap.node[[].spi are kept consistent.

Reply:
@suwang48404 if we revert the `append` we will not be able to find the `key` in the slice when we get an update to delete the `key`, resulting in a similar error  :)